### DISCUSSION TITLE
Use client local authentication

### DISCUSSION THREAD

**Discussion by SimStone** [ltsp/ltsp#570] [Q&A] [answered]
Hello,

In my company we currently provide users access to xubuntu 16.04 diskless fat-clients using LTSP5 and PXE boot, the image being served with NBD. A xubuntu VM is used to create the image which is served using a debian server.
In this setup, we get clients to automatically login using an account created on the xubuntu VM by enabling AutomaticLogin options in /etc/gdm3/custom.conf (the server takes no part in authentication which is done only using the image files).
In the new LTSP we get autologin to work using the options in ltsp.conf but it means sshfs usage. Is there a way to do the authentication using only the image files ?

Best regards,

**Comment by alkisg** [comment]
Please login to an LTSP5 client, run the following commands, and upload the output here:

```shell
mount
df -h
lsblk --fs
blkid
```

By default the setup you're mentioning provides no writable /home, so I want to see what you're using for /home, possibly a bit of the tmpfs that LTSP used for overlay.

Autologin in the new LTSP can work with any /home, either sshfs, nfs, local partitions or RAM tmpfs, so sure it can do the same things that LTSP5 could, and more.

**Reply by SimStone** [reply]
Here is the output :

root@ltsp-10-209-1-104:~# mount
sysfs on /sys type sysfs (rw,nosuid,nodev,noexec,relatime)
proc on /proc type proc (rw,nosuid,nodev,noexec,relatime)
udev on /dev type devtmpfs (rw,nosuid,relatime,size=1937676k,nr_inodes=484419,mode=755)
devpts on /dev/pts type devpts (rw,nosuid,noexec,relatime,gid=5,mode=620,ptmxmode=000)
tmpfs on /run type tmpfs (rw,nosuid,noexec,relatime,size=393752k,mode=755)
/dev/nbd0 on /rofs type squashfs (ro,relatime)
tmpfs on /cow type tmpfs (rw,relatime,mode=755)
overlay on / type overlay (rw,relatime,lowerdir=/rofs,upperdir=/cow/up,workdir=/cow/work)
securityfs on /sys/kernel/security type securityfs (rw,nosuid,nodev,noexec,relatime)
tmpfs on /dev/shm type tmpfs (rw,nosuid,nodev)
tmpfs on /run/lock type tmpfs (rw,nosuid,nodev,noexec,relatime,size=5120k)
tmpfs on /sys/fs/cgroup type tmpfs (ro,nosuid,nodev,noexec,mode=755)
cgroup on /sys/fs/cgroup/systemd type cgroup (rw,nosuid,nodev,noexec,relatime,xattr,release_agent=/lib/systemd/systemd-cgroups-agent,name=systemd)
pstore on /sys/fs/pstore type pstore (rw,nosuid,nodev,noexec,relatime)
cgroup on /sys/fs/cgroup/rdma type cgroup (rw,nosuid,nodev,noexec,relatime,rdma)
cgroup on /sys/fs/cgroup/freezer type cgroup (rw,nosuid,nodev,noexec,relatime,freezer)
cgroup on /sys/fs/cgroup/net_cls,net_prio type cgroup (rw,nosuid,nodev,noexec,relatime,net_cls,net_prio)
cgroup on /sys/fs/cgroup/memory type cgroup (rw,nosuid,nodev,noexec,relatime,memory)
cgroup on /sys/fs/cgroup/cpu,cpuacct type cgroup (rw,nosuid,nodev,noexec,relatime,cpu,cpuacct)
cgroup on /sys/fs/cgroup/devices type cgroup (rw,nosuid,nodev,noexec,relatime,devices)
cgroup on /sys/fs/cgroup/pids type cgroup (rw,nosuid,nodev,noexec,relatime,pids)
cgroup on /sys/fs/cgroup/perf_event type cgroup (rw,nosuid,nodev,noexec,relatime,perf_event)
cgroup on /sys/fs/cgroup/cpuset type cgroup (rw,nosuid,nodev,noexec,relatime,cpuset)
cgroup on /sys/fs/cgroup/blkio type cgroup (rw,nosuid,nodev,noexec,relatime,blkio)
cgroup on /sys/fs/cgroup/hugetlb type cgroup (rw,nosuid,nodev,noexec,relatime,hugetlb)
systemd-1 on /proc/sys/fs/binfmt_misc type autofs (rw,relatime,fd=24,pgrp=1,timeout=0,minproto=5,maxproto=5,direct,pipe_ino=17071)
debugfs on /sys/kernel/debug type debugfs (rw,relatime)
hugetlbfs on /dev/hugepages type hugetlbfs (rw,relatime,pagesize=2M)
mqueue on /dev/mqueue type mqueue (rw,relatime)
fusectl on /sys/fs/fuse/connections type fusectl (rw,relatime)
configfs on /sys/kernel/config type configfs (rw,relatime)
tmpfs on /run/user/1099 type tmpfs (rw,nosuid,nodev,relatime,size=393752k,mode=700,uid=1099,gid=1000)
gvfsd-fuse on /run/user/1099/gvfs type fuse.gvfsd-fuse (rw,nosuid,nodev,relatime,user_id=1099,group_id=1000)
tmpfs on /run/user/0 type tmpfs (rw,nosuid,nodev,relatime,size=393752k,mode=700)

root@ltsp-10-209-1-104:~# df -h
Sys. de fichiers Taille Utilisé Dispo Uti% Monté sur
udev               1,9G       0  1,9G   0% /dev
tmpfs              385M    6,2M  379M   2% /run
/dev/nbd0          2,1G    2,1G     0 100% /rofs
tmpfs              1,9G    6,7M  1,9G   1% /cow
overlay            1,9G    6,7M  1,9G   1% /
tmpfs              1,9G       0  1,9G   0% /dev/shm
tmpfs              5,0M       0  5,0M   0% /run/lock
tmpfs              1,9G       0  1,9G   0% /sys/fs/cgroup
tmpfs              385M     28K  385M   1% /run/user/1099
tmpfs              385M       0  385M   0% /run/user/0

root@ltsp-10-209-1-104:~# lsblk --fs
NAME FSTYPE LABEL UUID MOUNTPOINT
nbd0                                    /rofs

root@ltsp-10-209-1-104:~# blkid
/dev/nbd0: TYPE="squashfs"

**Comment by alkisg** [comment]
The key line is: `tmpfs 1,9G 6,7M 1,9G 1% /cow`
Which means that your client /home uses up to 1.9 GB RAM and gets discarded on reboot.
That implies that up to 1.9 GB less RAM is available to the clients for real use, for opening programs.
I would suggest to use the default SSHFS with autologin if you want security, or NFS with autologin if you don't care about security.
With NFS you don't even need accounts on the server. Or you can use a single account for all clients, and share a different folder to each client.

If you want to keep using the client RAM, you can put SSHFS=0 in ltsp.conf, and AUTOLOGIN=username, and it should work without having to mount anything.

**Reply by SimStone** [reply]
We wish to keep using the client RAM so I tried to put SSHFS=0 in ltsp.conf and did a ltsp initrd but after that when I start a client to test the /home is still sshfs mounted

**Comment by alkisg** [comment]
What's the output of `dpkg -l ltsp` on the server, and  what on the client?

**Comment by SimStone** [comment]
I attached the two output in the following files, first is the server and then the client :
![dpkg_ltsp_server](https://user-images.githubusercontent.com/91613751/137139633-4cea9365-ca6d-4275-a963-00304322b784.PNG)
![dpkg_ltsp_client](https://user-images.githubusercontent.com/91613751/137139641-8267ad3d-ebae-4851-a442-be70deee9f48.PNG)
s

**Comment by alkisg** [comment, marked as answer]
Hmm, could you please test this instead?

```shell
POST_INIT_DISABLE_SSHFS="echo SSHFS=0 >>/etc/ltsp/pamltsp.conf"
```

**Reply by SimStone** [reply]
It works and is exactly what I needed.
Thanks a lot.
